src/ZkFold/Symbolic/Data/Ed25519.hs

```diff
@@ -30,17 +31,17 @@ import           ZkFold.Symbolic.Data.Eq
 import           ZkFold.Symbolic.Data.UInt
 import           ZkFold.Symbolic.Interpreter
 
-zpToEd :: Finite (Zp p) => Point (Ed25519 (Interpreter (Zp p))) -> Point (Ed25519 Void)
+zpToEd :: (Symbolic (Interpreter (Zp p)), FromConstant (Interpreter (Zp p) Par1) Natural ) => Point (Ed25519 (Interpreter (Zp p))) -> Point (Ed25519 Void)
```

Comment:
I feel like we do not have a `FromConstant (Interpreter ...) Natural` instance (and interpreter is not constant, so it would be strange to call it that)